connectors/tests/test_source.py

```diff
@@ -106,6 +109,346 @@ def test_get_source_klass_dict():
+            {
+                "port": {
+                    "type": "int",
+                    "value": 5,
+                    "validations": [
+                        {"type": ValidationTypes.GREATER_THAN.value, "constraint": 0},
+                        {"type": ValidationTypes.LESS_THAN.value, "constraint": 10},
+                    ],
+                }
+            }
+        ),
+        (
+            # list_type str
+            {
+                "option_list": {
+                    "type": "list",
+                    "value": ["option1", "option2", "option3"],
+                    "validations": [
+                        {"type": ValidationTypes.LIST_TYPE.value, "constraint": "str"},
+                    ],
+                }
+            }
+        ),
+        (
+            # list_type int
+            {
+                "option_list": {
+                    "type": "int",
+                    "value": [1, 2, 3],
+                    "validations": [
+                        {"type": ValidationTypes.LIST_TYPE.value, "constraint": "int"},
+                    ],
+                }
+            }
+        ),
+        (
+            # included_in when list
+            {
+                "option_list": {
+                    "type": "list",
+                    "value": ["option1", "option2"],
+                    "validations": [
+                        {
+                            "type": ValidationTypes.INCLUDED_IN.value,
+                            "constraint": ["option1", "option2"],
+                        },
+                    ],
+                }
+            }
+        ),
+        (
+            # included_in when str
+            {
+                "option_str": {
+                    "type": "str",
+                    "value": "option2",
+                    "validations": [
+                        {
+                            "type": ValidationTypes.INCLUDED_IN.value,
+                            "constraint": ["option1", "option2"],
+                        },
+                    ],
+                }
+            }
+        ),
+        (
+            # included_in when int
+            {
+                "option_int": {
+                    "type": "int",
+                    "value": 2,
+                    "validations": [
+                        {
+                            "type": ValidationTypes.INCLUDED_IN.value,
+                            "constraint": [1, 2],
+                        },
+                    ],
+                }
+            }
+        ),
+        (
+            # regex
+            {
+                "email": {
+                    "type": "str",
+                    "value": "real@email.com",
+                    "validations": [
+                        {
+                            "type": ValidationTypes.REGEX.value,
+                            "constraint": "([A-Za-z0-9]+[.-_])*[A-Za-z0-9]+@[A-Za-z0-9-]+(\\.[A-Z|a-z]{2,})+",
```

Comment:
nit: The email regex could be a constant I guess as it's reused at multiple places.

Reply:
Yeah this will be especially needed as we expand it to other connectors. I'll include it in the next PR that adds these validations to connectors.